identity/test/services/MembersDataApiServiceTest.scala

```diff
@@ -0,0 +1,130 @@
+package services
+
+import conf.IdentityConfiguration
+import org.scalatest.mockito.MockitoSugar
+import org.scalatest.AsyncFlatSpec
+import org.scalatest.Matchers._
+import org.mockito.Mockito._
+import org.mockito.Matchers._
+import play.api.libs.ws.{WSClient, WSRequest, WSResponse}
+import play.api.mvc.{Cookie, Cookies}
+import play.api.libs.json.Json
+import scala.concurrent.Future
+
+class MembersDataApiServiceTest
+  extends AsyncFlatSpec
+  with MockitoSugar {
+
+  val mdapiApiRoot = "https://mdapimock.com"
+  val testUserId = "10000001"
+
+  val valid404ResponseJsonString =
+    """
+      |{
+      |  "message":"Not found",
+      |  "details":"Could not find user in the database",
+      |  "statusCode":404
+      |}
+    """.stripMargin
+
+  val validMdapiResponseJsonString =
+    """
+      |{
+      |"userId": "10000001",
+      |"contentAccess": {
+      |"member": true,
+      |"paidMember": true,
+      |"recurringContributor": false,
+      |"digitalPack": true,
+      |"paperSubscriber": false,
+      |"guardianWeeklySubscriber": true
+      |}
+      |}
+    """.stripMargin
+
+  val invalidMdapiResponseJsonString =
+    """
+      |{
+      |"userId": "10000001",
+      |"contentAccess": {
+      |"member": true,
+      |"paidMember": true,
+      |"digitalPack": true,
+      |"paperSubscriber": false,
+      |"guardianWeeklySubscriber": true
+      |}
+      |}
+    """.stripMargin
+
+
+  private val config = mock[IdentityConfiguration]
+  private val cookies = Cookies(Seq(Cookie("Cookie1", "hash", None, "/", None, false, true, None), Cookie("Cookie2", "hash", None, "/", None, false, true, None)))
+
+  private val wsThrow = mock[NullPointerException]
+  private val wsMock = mock[WSClient]
+  private val wsRequestMock = mock[WSRequest]
+  private val wsResponseMock = mock[WSResponse]
+  private val MdapiService = new MembersDataApiService(wsMock, config)
+
+  when(wsMock.url(anyString())).thenReturn(wsRequestMock)
+  when(wsRequestMock.withCookies(any())).thenReturn(wsRequestMock)
+
+  "getUserContentAccess" should
+    "return ContentAccess data for user with valid cookies" in {
+    when(wsRequestMock.get()).thenReturn(Future.successful(wsResponseMock))
+    when(wsResponseMock.json).thenReturn(Json.parse(validMdapiResponseJsonString))
+    when(wsResponseMock.status).thenReturn(200)
+    when(wsResponseMock.statusText).thenReturn("OK")
+
+    val futureResult = MdapiService.getUserContentAccess(cookies)
+    futureResult map {
+      result =>
+        result.fold(l => println(s"left error:${l.message}"), r => (println(s"right: $r")))
```

Comment:
No need to log from tests. Test framework's message in terms of test name, matcher-message and stack trance should be good enough.